src/core/structure.js

```diff
@@ -223,6 +224,30 @@ function getNonintroductorySectionHeaders() {
   );
 }
 
+/**
+ * For all sections having `data-max-toc=N`, make sure all their child sections
+ * do not end up in ToC.
+ */
+function skipFromToC() {
+  /** @type {NodeListOf<HTMLElement>} */
+  const sections = document.querySelectorAll("section[data-max-toc]");
+  for (const section of sections) {
+    const maxToc = parseInt(section.dataset.maxToc, 10) + 1;
+    if (maxToc < 0 || maxToc > 6) {
+      continue; // TODO: pub(error)
+    }
+    // With `data-max-toc=0`, only current section gets into ToC.
+    // With `data-max-toc=1`, we skip all ":scope > section > section" from ToC
+    // i.e., at §1, we will keep §1.1 but not §1.1.1
```

Comment:
Not sure if we want this behavior, or make `data-max-toc` equivalent to `.notoc` on current section (±1 depth).

Reply:
yeah, data-max-toc=0 === notoc would be less confusing.